### ISSUE TITLE
[BUG] A valid browser not found when debugging from local DevContainer

### ISSUE THREAD

**Issue by FBoucher** [microsoft/vscode-dotnettools#742]
### Describe the Issue

Debugging a web app in local DevContainer gives errors and does not allow for debugging.

![image](https://github.com/microsoft/vscode-dotnettools/assets/2404846/82861a37-9add-4dd3-9138-705eeb426395)

I had the same experience on a Windows 11 and Ubuntu 22.04.
It works fine when running locally (Edge is installed).
When I re-open in a devContainer I have the same "A valid browser is not installed." message.

### Steps To Reproduce

1. Create a new blazor wasm
2. reopen in a DEvContainer

devcontainer.json
```json

{
    "name": "C# (.NET) and MySQL",
    "dockerComposeFile": "docker-compose.yml",
    "service": "devbox",
    "workspaceFolder": "/workspaces/${localWorkspaceFolderBasename}"
	, "customizations": {
		"vscode": {
			"extensions": [
				"ms-dotnettools.csharp",
				"ms-dotnettools.csdevkit",
				"ms-dotnettools.blazorwasm-companion",
				"eamodio.gitlens"
			]
		}
	}
	, "forwardPorts": [5000, 5001, 3306]
	, "portsAttributes": {
			"5001": {
				"protocol": "https"
			}
	}
	, "remoteUser": "root"
}
```

dockerfile
```dockerfile
FROM mcr.microsoft.com/devcontainers/dotnet:8.0
```

### Expected Behavior

Debug launches with the app opening in the browser

### Environment Information

- OS:  Windows 11 and Ubuntu 22.04
- x64
- dotnet --version   8.0.100
- Extension
  - C# DevKit v1.0.4
  - C# v2.3.27

**Comment by timheuer** [comment]
@FBoucher I don't think this is a dev kit issue, but rather needing dependencies in the devcontainer you are building to satisfy the scenarios.

**Comment by timheuer** [comment]
@WardenGnaw I'm not sure if these errors are coming from vsdbg or the companion.  In 'normal' web debug even in a devcontainer int launches the host browser and tunnels. In wasm, it seems that assumption is not made?

Repro project: https://github.com/timheuer/csdevkit-742

**Comment by WardenGnaw** [comment]
@timheuer This error is from the C# Extension. Specifically https://github.com/dotnet/vscode-csharp/blob/e5c352d80af74199b832404cfe2b8e93809e84bf/src/razor/src/blazorDebug/blazorDebugConfigurationProvider.ts#L182

@phil-allen-msft Does Blazor Debugging work with dev containers?

**Comment by FBoucher** [comment]
@phil-allen-msft is it supported?

**Comment by MSFTJim** [comment]
I am also getting these error messages.  I get them when I try to use the C# DevKit and debug on a Blazor project that includes wasm (wasm or auto).  If the project does not include wasm, then the devkit debug works fine.  If the project does include wasm, I discovered I could just create the launch and task entries and then I can debug.  I was doing this with Codespaces.

Not sure if this helps but thought I would share.

**Comment by WellspringCS** [comment]
I see these errors, too, basically the same setup. Dev container, blazor.

**Comment by dkwgit** [comment]
Not sure whom to ask, perhaps @phil-allen-msft , but it would be good to have confirmed if this is currently not working, since that would save the trouble of trying to make it work.

**Comment by FBoucher** [comment]
Let me know if I can help validating or something. Happy to help.

**Comment by clarity99** [comment]
I get this on a macos host, blazor server project.

**Comment by serefarikan** [comment]
My case is using a devcontainer over SSH and I'm hitting the same problem. I ssh into WSL, then start a devcontainer and attempting to debug blazor ends up with this error displayed by vs code. It sounds like SSH is not the factor here, vs code extension is simply not aware of the container context.

**Comment by macel94** [comment]
> [!CAUTION]
> If you do not want to lose time as i did, do not try to continue with this.

If you install edge as i did both in wsl and in my devcontainer
```
## Edge Setup - needed for wasm debugging
curl https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor > microsoft.gpg
sudo install -o root -g root -m 644 microsoft.gpg /usr/share/keyrings/
sudo sh -c 'echo "deb [arch=amd64 signed-by=/usr/share/keyrings/microsoft.gpg] https://packages.microsoft.com/repos/edge stable main" > /etc/apt/sources.list.d/microsoft-edge-beta.list'
sudo rm microsoft.gpg

## Install
sudo apt update
sudo apt install microsoft-edge-beta

```
then you can launch the debugger, but you will fail anyway.
This is because debugging is not officially supported in wsl or devcontainers:
https://learn.microsoft.com/en-us/aspnet/core/blazor/debug?view=aspnetcore-9.0&tabs=visual-studio-code

> Available scenarios for Blazor WebAssembly debugging include:
> 
> Set and remove breakpoints.
> Run the app with debugging support in IDEs.
> Single-step through the code.
> Resume code execution with a keyboard shortcut in IDEs.
> In the Locals window, observe the values of local variables.
> See the call stack, including call chains between JavaScript and .NET.
> Use a [symbol server](https://learn.microsoft.com/en-us/aspnet/core/test/debug-aspnetcore-source?view=aspnetcore-9.0) for debugging, configured by Visual Studio preferences.
> Unsupported scenarios include:
> 
> Debug in non-local scenarios (for example, [Windows Subsystem for Linux (WSL)](https://learn.microsoft.com/en-us/windows/wsl/) or [Visual Studio Codespaces](https://visualstudio.microsoft.com/services/github-codespaces/)).
> Debug in Firefox from Visual Studio or Visual Studio Code.
> 

also:

> Start Without Debugging [Ctrl+F5 (Windows) or ⌘+F5 (macOS)] isn't supported. When the app is run in Debug configuration, debugging overhead always results in a small performance reduction.

**Comment by serefarikan** [comment]
@macel94 you're right about debugging not being supported. However, vs code does a great job of launching a local (as in on the machine running vs code) browser and redirecting the ports in many cases. Installing a browser in WSL or the container is a workaround, but still significantly inferior to convenience vs code provides in other cases.

**Comment by hades200082** [comment]
I'm seeing this same error when launching Blazor via an Aspire 9 project too.

**Comment by ryzngard** [comment]
For "remote" cases like WSL, DevContainer, Codespaces, etc we're tracking with https://github.com/microsoft/vscode-dotnettools 

Closing as a dupe for now. If you're hitting a scenario that is not covered in that list please file a new bug.
